lispyville.el

```diff
@@ -665,6 +665,74 @@ ARG has the same effect."
       (lispy--normalize-1))
     (goto-char orig-pos)))
 
+(evil-define-operator lispyville-comment-or-uncomment (beg end type)
+  "Like `evilnc-comment-operator' but handles unmatched delimiters.
+May insert newlines where necessary in order to move delimiters outside of the
+commented regions and preserve the balanced structure. "
+  (interactive "<R>")
+  (when (string-match-p (format "[%s]+$" (regexp-quote comment-start))
```

Comment:
Could you explain the purpose of this check? Also, `comment-start` is not always a single character (e.g. in c-mode it is `/* `); I'm guessing `comment-start-skip` should be used instead.

Reply:
This was due to a strange issue I encountered with undo-tree that would happen in the following situations:
- With the cursor in front of multiple comment characters, attempt to comment-or-uncomment only one character ("gcl")
```
|;; some comment
```
Result: the entire line is uncommented, but all previous undo history in the buffer gets completely mangled. 
Trying to `undo-tree-undo` produces the error:
```
primitive-undo: changes to function different than announced
```
- attempting to comment a word in the middle of a line:
```
foo |bar baz

;; "gce" ->

foo ;; bar
baz
```
but corrupts undo history as well, explaining the check slightly further down.

After spending way too long trying to dig into the root cause (appears to be an upstream bug with either `uncomment-region` or `undo-tree`) I ended up just adding the above as workarounds 😕

Reply:
That's fine for now, but hopefully this can be fixed in `undo-tree` and removed in the future. Can you add a similar comment above this check too explaining that it's a workaround for `undo-tree`? I've stopped using `undo-tree` entirely because of corruption issues personally.